serializers/src/glTF/2.0/Extensions/KHR_lights_punctual.ts

```diff
@@ -0,0 +1,185 @@
+        }
+    }
+    public postExportNodeAsync(context: string, node: INode, babylonNode: Node): Nullable<Promise<INode>> {
+        return new Promise((resolve, reject) => {
+            if (babylonNode instanceof ShadowLight) {
+                let babylonLight: ShadowLight = babylonNode;
+                let light: ILight;
+
+                let lightType = (
+                    babylonLight.getTypeID() == Light.LIGHTTYPEID_POINTLIGHT ? LightType.POINT : (
+                        babylonLight.getTypeID() == Light.LIGHTTYPEID_DIRECTIONALLIGHT ? LightType.DIRECTIONAL : (
+                            babylonLight.getTypeID() == Light.LIGHTTYPEID_SPOTLIGHT ? LightType.SPOT : null
+                        )));
+                if (lightType == null) {
+                    Tools.Warn(`${context}: Light ${babylonLight.name} is not supported in {NAME}`);
+                }
+                else {
+                    let lightPosition = babylonLight.position.clone();
+                    if (!lightPosition.equalsToFloats(1.0, 1.0, 1.0)) {
+                        if (this._exporter._convertToRightHandedSystem) {
+                            lightPosition.z *= -1;
+                        }
+                        node.translation = lightPosition.asArray();
+                    }
+                    if (babylonLight.falloffType != Light.FALLOFF_GLTF) {
+                        Tools.Warn(`${context}: Light falloff for ${babylonLight.name} does not match the ${NAME} specification!`);
+                    }
+                    light = {
+                        type: lightType
+                    };
+                    if (!babylonLight.diffuse.equalsFloats(1.0, 1.0, 1.0)) {
+                        light.color = babylonLight.diffuse.asArray();
+                    }
+                    if (!(babylonLight.intensity == 1.0)) {
+                        light.intensity = babylonLight.intensity;
+                    }
+                    if (!(babylonLight.range == Number.MAX_VALUE)) {
+                        light.range = babylonLight.range;
+                    }
+
+                    if (lightType === LightType.SPOT) {
+                        let babylonSpotLight = babylonLight as SpotLight;
+                        if (!(babylonSpotLight.angle == Math.PI / 2.0)) {
+                            if (light.spot == null) {
+                                light.spot = {};
+                            }
+                            light.spot.outerConeAngle = babylonSpotLight.angle / 4;
+                        }
+                        if (!(babylonSpotLight.innerAngle == 0)) {
+                            if (light.spot == null) {
+                                light.spot = {};
+                            }
+                            light.spot.innerConeAngle = babylonSpotLight.innerAngle / 2;
+                        }
+                        let rotation = babylonSpotLight.getRotation().clone();

```

Comment:
Can be shared with the code below? `getRotation` is on `ShadowLight`

Reply:
@bghgary I can bump it up. I wasn't sure if it made sense to have it on PointLight, but it should just be ignored by engines and can save code duplication.

Reply:
You can always put another if check around it.

Reply:
@bghgary actually, I was handling rotation incorrectly.  I will make an update to use the direction on the spotlight and directional light instead.

Reply:
update has been made